spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/PropertyMapper.java

```diff
@@ -308,6 +310,21 @@ public <R> R toInstance(Function<T, R> factory) {
 			return factory.apply(value);
 		}
 
+		/**
+		 * Complete the mapping by creating a new instance from the non-filtered value.
+		 * @param <R> the resulting type
+		 * @param factory the factory used to create the instance
+		 * @return the created instance or empty when the value is filtered
+		 */
+		public <R> Optional<R> toOptionalInstance(Function<T, R> factory) {
```

Comment:
I don't think this description is quite right, specifically "from the non-filtered value" seems to be contradicted by "or empty when the value is filtered". I think the opening sentence should end with "from the filtered value".

Given the use of the term `filtered` here and in the javadoc of various other methods, I think `toFilteredInstance` would be a better name for the method.

Reply:
@wilkinsona I too thought the terms were off but in the PropertyMapper the term "filtered" means that it did not pass all of the predicates - unfiltered means it passed all of the predicates. Is that correct @philwebb ?

Reply:
On second reading, you're right I think. Perhaps just "Complete the mapping by creating a new instance" would be better?

That leaves me unsure about the method name. I'm still not keen on `toOptionalInstance`. Maybe `toFilteredInstance` is still ok. It's an instance created from a value that has gone through filtering. As a result of that filtering it may be `null` or it may be the value 🤷‍♂️

Reply:
Sounds good on the doc update. 

> That leaves me unsure about the method name. I'm still not keen on toOptionalInstance

For consistency we would then need probably need to update the existing `toInstance`. Going a little further, would we also add "filtered" and "non-filtered" to the other APIs that only "do their thing" (call, consume, etc) when they are non-filtered?

Reply:
I don't like `toOptionalInstance` either. I wonder if we shouldn't keep `toInstance` in both cases but with a boolean flag that express what should happen if the predicate does not match.

I should say that, out-of-context, the "non-filtered" bit is a bit hard to grasp.

Reply:
@snicoll I pushed a commit w/ the suggestion. I am ok w/ either approach. 

> I should say that, out-of-context, the "non-filtered" bit is a bit hard to grasp.

Agreed